message_ix_models/model/material/data_util.py

```diff
@@ -2223,3 +2216,33 @@ def add_elec_i_ini_act(scenario: message_ix.Scenario) -> None:
     scenario.add_par(par, df_el)
     scenario.commit("add initial_activity_up for elec_i")
     return
+
+
+def calculate_ini_new_cap(df_demand, technology, material):
```

Comment:
```suggestion
def calculate_ini_new_cap(df_demand: pd.DataFrame, technology: str, material: str) -> pd.DataFrame:
```
This would make the function more beautiful, though it might require running ruff again.